tfjs-backend-webgpu/src/kernels/depth_to_space_webgpu.ts

```diff
@@ -0,0 +1,140 @@
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  workGroupSize: [number, number, number] = [64, 1, 1];
+  size: number;
+  useWgsl: boolean;
+  uniforms = 'int blockSize;';
+  uniformsWgsl = 'blockSize : u32;';
+
+  constructor(outputShape: number[], dataFormat: 'NHWC'|'NCHW') {
+    this.outputShape = outputShape;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workGroupSize);
+    this.shaderKey = `depthToSpace_${dataFormat}`;
+    this.size = util.sizeFromShape(this.outputShape);
+    this.dataFormat = dataFormat;
+  }
+
+  getUserCode(): string {
+    const userCode = `
+      void main() {
+        int index = getGlobalIndex();
+        if (index < size) {
+          ivec4 coords = getOutputCoords();
+          int b = coords[0];
+          int h = ${this.getHeightCoordString()};
+          int w = ${this.getWidthCoordString()};
+          int d = ${this.getDepthCoordString()};
+
+          int in_h = h / blockSize;
+          int offset_h = h % blockSize;
+          int in_w = w / blockSize;
+          int offset_w = w % blockSize;
+          int offset_d = (offset_h * blockSize + offset_w) *
+            ${this.getOutputDepthSize()};
+          int in_d = d + offset_d;
+
+          float result = ${this.getInputSamplingString()};
+          setOutput(index, result);
+        }
+      }`;
+    return userCode;
+  }
+
+  getUserCodeWgsl(): string {
+    const userCode = `
+      ${getMainHeaderStringWgsl()} {
+        ${getGlobalIndexStringWgsl()}
+        let index = getGlobalIndex();
+        if (index < size) {
+          let coords = getOutputCoords();
+          let b = coords[0];
+          let h = ${this.getHeightCoordString()};
+          let w = ${this.getWidthCoordString()};
+          let d = ${this.getDepthCoordString()};
+
+          let in_h = h / uniforms.blockSize;
+          let offset_h = h % uniforms.blockSize;
+          let in_w = w / uniforms.blockSize;
+          let offset_w = w % uniforms.blockSize;
+          let offset_d = (offset_h * uniforms.blockSize + offset_w) *
+            ${this.getOutputDepthSize()};
+          let in_d = d + offset_d;
+
+          let result = ${this.getInputSamplingString()};
+          setOutput(index, result);
+        }
+      }`;
+    return userCode;
+  }
+
+  private getHeightCoordString(): string {
+    if (this.dataFormat === 'NHWC') {
+      return `coords[1]`;
+    } else {
+      return `coords[2]`;
+    }
+  }
+
+  private getWidthCoordString(): string {
+    if (this.dataFormat === 'NHWC') {
+      return `coords[2]`;
+    } else {
+      return `coords[3]`;
+    }
+  }
+
+  private getDepthCoordString(): string {
+    if (this.dataFormat === 'NHWC') {
+      return `coords[3]`;
+    } else {
+      return `coords[1]`;
+    }
+  }
+
+  private getOutputDepthSize(): string {
+    if (this.dataFormat === 'NHWC') {
+      return `outShape[3]`;
```

Comment:
This should be different for GLSL and WGSL. For WGSL, this shoule be:  uniforms.outShape[3]

Reply:
Done.